.clang-tidy

```diff
@@ -27,7 +25,43 @@
 #   flexible. We consider this negligible compared to the benefits of
 #   manipulating variables that are explicitly typed.
 #
-Checks: >
+# -modernize-use-trailing-return-type,
+# -modernize-return-braced-init-list,
+#   These are removed since they conflict with the current code style.
+#
+# -modernize-pass-by-value
+# -modernize-deprecated-headers
+# -modernize-use-default-member-init
+# -readability-redundant-member-init
+# -bugprone-exception-escape
+#   These conflict with the current code style, or would require larger
+#   changes. May wish to review in the future.
+#
+# -readability-magic-numbers
+#   Disabled for now as it picks up some constants which don't necessarily need
+#   their own variables (for some trivial multiplication and masking) and some
+#   test data. May wish to review in the future.
+#
+# -readability-implicit-bool-conversion
+#   We use implicit bool conversion for specifying many constants in terms of
+#   `0` and `1`. (TODO: it may be a good idea to enable this to fix implicit
```

Comment:
We can enable this check bu we will need to add a lot of `NOLINT` comments to disable the checks when we use pb_variable_arrays of booleans (like in blake/sha). This is what I did on https://github.com/clearmatics/zeth/pull/212